nexus/reconfigurator/planning/src/blueprint_builder/builder.rs

```diff
@@ -1435,6 +1689,338 @@ impl<'a> BlueprintDisksBuilder<'a> {
     }
 }
 
+/// Helper for working with sets of datasets on each sled
+struct BlueprintDatasetsBuilder<'a> {
```

Comment:
Would it make sense to move these new types into a submodule? (I find this file kinda unwieldy personally, so just looking for ways to keep it from growing more.)

Reply:
I think this should happen, but are you okay if I punt this to a follow-up PR to help the diff remain reasonable here? I also think there are some non-dataset things which should be submodule'd.

Reply:
Certainly :+1: